### ISSUE TITLE
Proto file sourcing

### ISSUE THREAD

**Issue by SkinnySackboy** [tensorflow/serving#2228]
Hi all,

I'm fairly new to TF Serving (I've been tasked with making calls to it). For some background here, we are interested in making TF Serving calls from .NET.

Usually we are used to easily sourcing the *.proto files, generating a client automatically within .NET, job done. 

With TF Serving, we are having a hard time identifying where to get these files from. Aside from there being plenty of files (not an issue) they seem to be spread across multiple places. Furthermore, they seem to be mixed in folders containing code, and some of these files seem to reference *.proto files in entirely separate github repositories.

As such, it has become incredibly laborious for us to identify these proto files and organize them in such a way that we can automatically generate our clients. Furthermore, upon successive versions, backwards compatibility aside, it is rather painful to perform this process from scratch.

Am I doing something wrong? Is there a more effective place to get the *.proto files from that I'm not aware of?

Thanks in advance.

**Comment by quantike** [comment]
Curious if you've found any solutions here? I am tasked with a similar issue and cannot seem to get the *.proto files to correctly compile. There is seemingly an issue with all the `error_codes.proto` files. There is some transition happening that is moving those files around.

**Comment by quantike** [comment]
Ouch. Hopefully there is an official tool eventually, or at least a link to something that is known to work

**Comment by SkinnySackboy** [comment]
Agreed, it's remarkable that hardly anyone else seems to be complaining about this. It's literally the entire point of gRPC.

**Comment by janasangeetha** [comment]
Hi @SkinnySackboy 
Sorry for responding this issue late. Usually `*.proto` files are stored in `tensorflow_serving/apis/*.proto`.
Optionally you can use [Buf’s managed mode](https://github.com/bufbuild/buf-examples/tree/main/managed-mode) to get all the `*.proto` files in one folder.
Please feel free to reach out to us if you need further information.
Thank you!

**Comment by SkinnySackboy** [comment]
Thanks @janasangeetha I'll try this out as soon as I get the chance. Last time I did though, the proto files were referencing other files from all over the place, including other GitHub repos. The bit I'm not clear on is, assuming you can get them all into a single folder, how this would work as any paths within the files themselves would still need to be fixed.

**Comment by janasangeetha** [comment]
Hi @SkinnySackboy 
Yes, the files has to be updated with paths. I would suggest to have all the paths defined in a common file so that it becomes easy to update and reuse.
Incase you come-up with any other solution, please feel free to share with us.
Thank you!
